src/bindings/python/flux/util.py

```diff
@@ -380,3 +382,163 @@ def format(self, obj):
             typestr = type(obj)
             raise KeyError(f"Invalid format field {exc} for {typestr}")
         return retval
+
+
+class Tree:
+    """Very simple pstree-like display for the console
+
+    Args:
+        label: Label for this node
+        prefix: Text which comes before connector and label
+        combine_children: combine like children as they are added
+    """
+
+    Connectors = namedtuple("Connectors", ["PIPE", "TEE", "ELBOW", "BLANK"])
+
+    connector_styles = {
+        "ascii": Connectors("|   ", "|-- ", "`-- ", "    "),
+        "box": Connectors("│   ", "├── ", "└── ", "    "),
+        "compact": Connectors("│ ", "├─", "└─", "  "),
+    }
+
+    def __init__(self, label, prefix="", combine_children=False):
+
+        self.label = label
+        self.prefix = prefix
+        if self.prefix:
+            self.prefix += " "
+
+        self.duplicates = {}
+        self.children = []
+        self.count = 1
```

Comment:
would suggest `combine_count` or `duplicate_count` or something like that for this variable, as I thought we might be counting levels or something.  Below

```
      if self.count > 1:
            label = f"{self.count}*[{self.label}]"
```

was initially confusing, I had no idea what was going on until I figured out "ohh this is the duplicate count label".

perhaps `increment` below would be renamed as well.

Reply:
That seems like a valid suggestion.

Reply:
fixed in most recent push